src/osa/scripts/gainsel_webmaker.py

```diff
@@ -0,0 +1,138 @@
+from pathlib import Path
+from astropy.table import Table
+from datetime import datetime, timedelta
+from osa.utils.utils import date_to_dir, date_to_iso
+from osa.configs.config import cfg
+from osa.paths import DEFAULT_CFG
+from osa.nightsummary.nightsummary import run_summary_table
+import numpy as np
+import pandas as pd
+from argparse import ArgumentParser
+from osa.configs import options
+from osa.scripts.sequencer_webmaker import html_content
+
+def valid_date(string):
+    """Check if the string is a valid date and return a datetime object."""
+    return datetime.strptime(string, "%Y%m%d")
+
+
+common_parser = ArgumentParser(add_help=False)
+common_parser.add_argument(
+    "-c",
+    "--config",
+    type=Path,
+    default=DEFAULT_CFG,
+    help="Use specific config file [default configs/sequencer.cfg]",
+)
+common_parser.add_argument(
+    "-d",
+    "--date",
+    help="Date (YYYYMMDD) of the start of the night",
+    type=valid_date,
+)
+
+
+def check_gainsel_jobs_runwise(date: str, run_id: int) -> bool:
+    """Search for failed jobs in the log directory."""
+    base_dir = Path(cfg.get("LST1", "BASE"))
```

Comment:
```suggestion
    base_dir = Path(cfg.get("LST1", "BASE"))
    flat_date = date_to_dir(date)
```

Reply:
The input would be flat YYYYMMDD already